orchestra/project.py

```diff
@@ -1,10 +1,17 @@
+import threading
```

Comment:
Curiously, are there any asynchrony concerns? For example, if a machine task runs immediately after project creation (as we have in our internal workflows) and needs the folder, will it try to access the folder before it's created?